syntaxes/bats.extended.tmLanguage

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+    <dict>
+        <key>scopeName</key>
+        <string>source.shell.bats.extended</string>
+        <key>injectionSelector</key>
+        <string>L:source.shell.bats -string -comment</string>
+        <key>patterns</key>
+        <array>
+            <dict>
+                <key>match</key>
+                <string>@test\b</string>
+                <key>name</key>
+                <string>keyword.control.bats</string>
+            </dict>
+            <!-- bats -->
+            <dict>
+                <key>match</key>
+                <string>\b(run|load|skip)\b</string>
+                <key>name</key>
+                <string>support.function.bats.builtin</string>
+            </dict>
+            <dict>
+                <key>match</key>
+                <string>\$(output|status|lines)\b</string>
```

Comment:
`$lines` is usually used as `"${lines[0]}", ${lines[0]}` or `"${lines[@]}`/ `"${lines[*]}` Could you add those as well?

I'm most worried about the quoted form, because `injectionSelector` excludes strings. Probably need another file...

Reply:
I will test how it is managed by VSC. I din't get time yesterday.
I guess the wildcard is the simplest solution if it's well managed by VSC.

Reply:
Just thinking about it.
The variables are already coloured by SH syntax without the `support.variable.bats` element.
I don't see the use-case for that. Could you provide an exemple ?
<img width="355" alt="Screenshot 2019-06-20 at 13 59 40" src="https://user-images.githubusercontent.com/6155133/59847856-fd2ad980-9363-11e9-98a7-c4009ad71a0b.png">

Reply:
the variables are colored as shell variables. But they have special meaning in bats. So I'd like to have `$output` colored differently than `$my_output`

Use cases:
- Easier visible
- Detects typos like `$uotput`

Reply:
I understand the use-cases but I'm not completely convinced for two reasons :

First, Bats is using a valid sh syntax.
The tree bats variables are just "reserved" variables names used by BATS.
So if I use shellcheck, the bats file should be valid (the green one are not on my screenshot, see next point).

Second, I've found no satisfying way to be able to manage the colouring if you use quotes or with any kind of brackets. (and I've tested many regex)
And that's logic so far as bats syntax is build on top of shell one, the regex is just apply on vars not on vars parameters or strings.
To manage that may should we create a new language definition... may actuals suff is enough so far ?
See bellow, just the two first ones are well coloured :
![image](https://user-images.githubusercontent.com/6155133/59880276-f5435780-93ac-11e9-8acd-d0b55c372f36.png)

Let me knew what you think or if you have additional informations on that topic.
Otherwise, we could probably close it or may this could stay as a new open issue requiring additional support.

Best.

Reply:
> The tree bats variables are just "reserved" variables names used by BATS.
So if I use shellcheck, the bats file should be valid (the green one are not on my screenshot, see next point).

Agreed. But why is this against this proposal? Highlighting the special variables is useful because they are special. Just the same as the functions.

I was able to find a way: A 2nd language injection. Remove the variables from the first (source.shell.bats.extended), then add another entry to package.json

                      {
				"scopeName": "source.shell.bats.variables",
				"path": "./syntaxes/bats.variables.tmLanguage",
				"injectTo": [ "source.shell.bats" ]
			}

And finally the new definition:

```
<?xml version="1.0" encoding="UTF-8" ?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
    <dict>
        <key>scopeName</key>
        <string>source.shell.bats.variables</string>
        <key>injectionSelector</key>
        <string>L:source.shell.bats -comment</string>
        <key>patterns</key>
        <array>
            <dict>
                <key>match</key>
                <string>\$(output|status|lines)\b</string>
                <key>name</key>
                <string>support.variable.bats</string>
            </dict>
            <dict>
                <key>match</key>
                <string>\$\{#?(output|status|lines)\b</string>
                <key>captures</key>
                <dict>
                    <key>1</key>
                    <dict>
                        <key>name</key>
                        <string>support.variable.bats</string>
                    </dict>
                </dict>
            </dict>
        </array>
    </dict>
</plist>
```

But I don't insist on that. Would be nice to have, but having the functions already helps

Reply:
Hi,

Well done, I didn't think about it !
It's not perfect but works.
![image](https://user-images.githubusercontent.com/6155133/59914183-273ed300-941a-11e9-8376-f1ed0d12048a.png)

So I've just push an update where I allow colour change on settings and I provided a sample setting in README.

Could you please test the updated module on your real files before I merge ?
Thanks

Reply:
Great! :)